### DISCUSSION TITLE
Becoming a Maintainer

### DISCUSSION THREAD

**Discussion by AlexDev404** [microsoft/winget-pkgs#79979] [General] [answered]
> https://github.com/microsoft/winget-pkgs/pull/62899#pullrequestreview-1001299350

Hi @ItzLevvie it's been a while,

I know it's a bit awkward posting this due to the fact that I asked this question 3 months ago. 
But let's just say there has been a couple IRL issues since then...

I now have a bit of free time on my hands though, which is why I wanted to ask if it's possible for me to **become a maintainer of the WinGet repository**? 

### Why Do I Want to Do This?

I've really open and wanting to do this for a while now because I like the idea of communities and giving back to people. 

### Why Do I Think I'm Suitable for This?

I'm great at following specifications, directions and love looking over what people have done and giving feedback to them.
<br/>
So what do you think Levvie? Is this a possibility? Are there any other requirements involved in becoming a maintainer? 
If so I'd like to know them. 

If you have any thoughts just leave them below because I'd love to hear them.
Thanks once again for being the one to review my PRs when I post them. 

Looking forward to discussing this with and hearing from you or one of the other maintainers,
AlexDev404

**Comment by vedantmgoyal9** [comment]
If you want to help the community and want to approve PRs, you might have to use the term, "Moderator".

Here, a maintainer is referred to as a person/organization who maintains a particular application/software/package, its versions, and metadata.

You can also get help at the chat: https://gitter.im/Microsoft/winget-pkgs.

Related Discussions:
- https://github.com/microsoft/winget-pkgs/discussions/15607 (this is already very cluttered so you might post this on the latter one)
- https://github.com/microsoft/winget-pkgs/discussions/25830

**Reply by AlexDev404** [reply]
Yes, by maintainer that's what I meant. But if moderator is a good starting point then I'd be willing to fill that position.

What is the process of becoming a moderator? Would you be willing to explain?

**Reply by vedantmgoyal9** [reply]
* https://github.com/microsoft/winget-pkgs/discussions/15607
* https://github.com/microsoft/winget-pkgs/discussions/25830
* https://github.com/microsoft/winget-pkgs/blob/master/Moderation.md

These 3 sources can be helpful for you to know what it takes to become a moderator. You can also ask @denelon, if you've any other questions related to becoming a moderator.

**Reply by AlexDev404** [reply]
> * [Moderation #15607](https://github.com/microsoft/winget-pkgs/discussions/15607)
> * [Additional Moderators #25830](https://github.com/microsoft/winget-pkgs/discussions/25830)
> * https://github.com/microsoft/winget-pkgs/blob/master/Moderation.md
> 
> These 3 sources can be helpful for you to know what it takes to become a moderator. You can also ask @denelon, if you've any other questions related to becoming a moderator.

Okay, I have taken a look at these. Thank you so much! I really appreciate it!

**Comment by denelon** [comment, marked as answer]
@alexDev404 welcome to the Windows Package Manager community! There are many different ways to contribute. These range from helping to support users who are reporting bugs or helping us to reproduce errors (if we don't have good steps) through to making product contributions in the form of Pull Requests (PRs) to implement Issues (features) or fixing bugs. And it's not limited to just those activities. 

There are many different facets to the product. We have repositories for generating manifests with the [Windows Package Manager Manifest Creator](https://github.com/microsoft/winget-create), submitting packages to the [Windows Package Manager Community Repository](https://github.com/microsoft/winget-pkgs), a reference implementation for a [REST source](htps://github.com/microsoft/winget-cli-restsource), and the [WinGet client](https://github.com/microsoft/winget-cli). There are also several tools utilities and documents distributed throughout these repositories.

There are no requirements to become a moderator or a maintainer in order to make an impact on the product or the catalog of packages available for customers. In fact, we've had several important contributions from the community at large. All we ask is to adhere to our [code of conduct](https://github.com/microsoft/winget-pkgs/blob/master/CODE_OF_CONDUCT.md) and the contributing guidelines (usually CONTRIBUTING.md in the root directory for each repository). Some of my favorite contributions have come from someone simply noticing a typo or an ambiguous sentence in documentation here at GitHub and at [Microsoft Docs](https://learn.microsoft.com/en-us/windows/package-manager/). Another important area is helping triage new Issues. Sometimes a duplicate Issue is being reported, and it helps to let the author know where to put their 👍 to help raise priority. Other times, there may already be a workaround or a solution to a problem.

We welcome constructive feedback on the product, the specifications, and PRs.

**Reply by AlexDev404** [reply]
> @AlexDev404 welcome to the Windows Package Manager community! There are many different ways to contribute. These range from helping to support users who are reporting bugs or helping us to reproduce errors (if we don't have good steps) through to making product contributions in the form of Pull Requests (PRs) to implement Issues (features) or fixing bugs. And it's not limited to just those activities.
> 
> There are many different facets to the product. We have repositories for generating manifests with the [Windows Package Manager Manifest Creator](https://github.com/microsoft/winget-create), submitting packages to the [Windows Package Manager Community Repository](https://github.com/microsoft/winget-pkgs), a reference implementation for a REST source, and the [WinGet client](https://github.com/microsoft/winget-cli). There are also several tools utilities and documents distributed throughout these repositories.
> 
> There are no requirements to become a moderator or a maintainer in order to make an impact on the product or the catalog of packages available for customers. In fact, we've had several important contributions from the community at large. All we ask is to adhere to our [code of conduct](https://github.com/microsoft/winget-pkgs/blob/master/CODE_OF_CONDUCT.md) and the contributing guidelines (usually CONTRIBUTING.md in the root directory for each repository). Some of my favorite contributions have come from someone simply noticing a typo or an ambiguous sentence in documentation here at GitHub and at [Microsoft Docs](https://learn.microsoft.com/en-us/windows/package-manager/). Another important area is helping triage new Issues. Sometimes a duplicate Issue is being reported, and it helps to let the author know where to put their 👍 to help raise priority. Other times, there may already be a workaround or a solution to a problem.
> 
> We welcome constructive feedback on the product, the specifications, and PRs.

Thank you so much @denelon ! Much appreciated! 
I have read these over and I agree to follow all of these to a tee. When can I start?

**Reply by denelon** [reply]
@AlexDev404 feel free to start contributing any time. Any moderators who are actively engaged with the project at the same time you are will likely take notice. The key to becoming nominated is based on how well you understand the product, the implications of making changes, and your interactions with the community. It's not something that will happen quickly. It will take time for others to observe your interactions with the community, and for you to demonstrate your knowledge. 

One of the best ways to make an impact is to help new users with issues, and to help PR authors with their submissions.
